src/js/components/Select/index.d.ts

```diff
@@ -32,10 +47,10 @@ export interface BasicSelectProps {
   margin?: MarginType;
   messages?: { multiple?: string };
   name?: string;
-  onChange?: (...args: any[]) => void;
-  onClose?: (...args: any[]) => any;
+  onChange?: (event: SelectOnChangeEvent) => void;
```

Comment:
do we also need `event?: SelectOnChangeEvent`

Reply:
I don' think so, because we already have onChange? and we have used onChange as a functional Prop to Select.

Reply:
I don't think that there will be a case when we use OnChange and don't have arguments to pass it to.

Reply:
This event type seems to be missing the actual event itself. When logging the event from `onChange`, the following event is fired depending on if I interact with mouse or keyboard.

<img width="710" alt="Screenshot 2024-10-29 at 3 01 00 PM" src="https://github.com/user-attachments/assets/287d51c9-bfef-4948-9896-8d285a66246e">
<img width="707" alt="Screenshot 2024-10-29 at 3 01 24 PM" src="https://github.com/user-attachments/assets/5dab2a7e-751e-4cf8-b34d-62ac198750b3">